src/entities/road.lua

```diff
@@ -86,10 +86,39 @@ function road:draw()
   love.graphics.setColor(255, 255, 255, 255)
 end
 
-function road:update(angle, roadShift)
+  
+function road:update(newSegments)
   -- create a road segment at the position and angle of the frontier.
-  self:addsegment()
+  for _, s in pairs(newSegments) do
+  self:addSegment(s.x, s.y, s.angle)
+  end
+  i = table.getn(newSegments)
+  self:setFrontier(newSegments[i].x, newSegments[i].y, newSegments[i].angle, "center")
+end
+
+
+-- utils
+
+function road:addSegment(x, y, angle)
+  segment = {}
+  segment.img = self.skin
+  segment.width = segment.img:getWidth()
+  segment.height = segment.img:getHeight()
+  segment.body = love.physics.newBody(world)
+  segment.body:setAngle(angle)
+  fx = math.sin(segment.body:getAngle()) * (segment.height/2)
+  fy = math.cos(segment.body:getAngle()) * -(segment.height/2)
+  segment.body:setX(x + fx)
+  segment.body:setY(y + fy)
+  segment.shape = love.physics.newRectangleShape(0, 0, segment.width, segment.height)
+  table.insert(self.segments, segment)
+end
+
+function road:getPaveThreshold()
+  return self.paveThreshold
+end
 
+function road:pushFrontier(angle, roadShift)
```

Comment:
this function can get merged above into `road:pushFrontier()` as it gets called every time from there, and it doesn't need to be called from anywhere else.

Reply:
see above comment